compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -974,7 +974,7 @@ TR::Register *OMR::X86::TreeEvaluator::baddEvaluator(TR::Node *node, TR::CodeGen
 
    if (NEED_CC(node))
       {
-      TR_ASSERT(node->getOpCodeValue() == TR::badd,
+      TR_ASSERT(node->getOpCodeValue() == TR::badd || node->getOpCodeValue() == TR::buadd,
                 "CC computation not supported for this node %p with opcode %s\n", node, cg->comp()->getDebug()->getName(node->getOpCode()));
 
       // we need eflags from integerAddAnalyser for the CC sequence
```

Comment:
Similar comment could be added for badd. The original comment which is still not very clear why the normal generation sequence has limitation.